### ISSUE TITLE
Pending Transaction From L2 to L1 Bridging

### ISSUE THREAD

**Issue by dc8791** [omgnetwork/optimism#565]
L2 to L1 fast bridging of OMG tokens is supposed to be fast. My transaction had been pending for 7 hours with 16 confirmations. What’s holding the pending status for so long?

**Comment by dc8791** [comment]
Transaction Hash
0xfcf9c2cabdcb54f1517eacda49090f948c14404587a0e14b7b995a90aebcf403

**Comment by CAPtheorem** [comment]
Please check now

**Comment by CAPtheorem** [comment]
Update - we just added a system to automatically flag stuck transactions and prioritize them, so this should stop happening. 

https://github.com/omgnetwork/optimism/pull/558#issue-1014568952

Root cause - in case you are interested. The underlying problem was the old logic that was used to pass messages when L1 gas was spiking. In that case, we waited for some time - e.g. 20 mins, in hope that gas would drop - but if it did not drop quickly enough, then pending transactions got stuck. In the new logic deployed around block 13361180, we now store those stuck transactions in a separate buffer and prioritize them for relay to L1.

Hope this helps!

Sorry for the TX delay - it's ironic that the 'fast' bridge was sometimes super slow.
